Typescript can not find declaration file TS7016

# Bug Report

## Relevant information

Typescript can not find a declaration file for module 'react-component-catalog' TS7016.

### Your Environment

* OS: Ubuntu 20.04
* Node: v12.20.2
* Package Version: 2.1.0

### Steps to reproduce

```shell
yarn create react-app my-app --template typescript
cd my-app
yarn add react-component-catalog
yarn start
```

Add ```import { useCatalog } from 'react-component-catalog';``` to any code file.
Typescript shows an error TS7016: Could not find a declaration file for module 'react-component-catalog'.

--- a/packages/react-component-catalog/package.json
+++ b/packages/react-component-catalog/package.json
@@ -23,17 +23,15 @@
     "lib"
   ],
   "scripts": {
-    "build": " yarn run build-types && yarn run build-cjs && yarn run build-es && yarn run build-esm && yarn run build-umd",
-    "build-types": "yarn run build-cjs-types && yarn run build-es-types && yarn run build-esm-types",
+    "build": "yarn run build-cjs && yarn run build-es && yarn run build-esm && yarn run build-umd",
     "build-cjs": "BABEL_ENV=cjs babel src --ignore **/*.test.tsx,**/*.test.ts --out-dir lib --extensions '.ts,.tsx'",
-    "build-cjs-types": "tsc -p tsconfig.build.json --outDir lib --module commonjs --target es5",
     "build-es": "BABEL_ENV=es babel src --ignore **/*.test.tsx,**/*.test.ts --out-dir es --extensions '.ts,.tsx'",
-    "build-es-types": "tsc -p tsconfig.build.json --outDir es --module es2015 --target es2015",
     "build-esm": "BABEL_ENV=esm babel src --ignore **/*.test.tsx,**/*.test.ts --out-dir esm --extensions '.ts,.tsx'",
-    "build-esm-types": "tsc -p tsconfig.build.json --outDir esm --module es2015 --target es5",
+    "build-types": "tsc -p tsconfig.build.json",
     "build-umd": "BABEL_ENV=umd webpack --mode=production",
     "coveralls": "jest --coverage && cat ./coverage/lcov.info | coveralls",
     "prebuild": "rimraf dist && rimraf es && rimraf esm && rimraf lib",
+    "postbuild": "yarn run build-types",
     "test": "tsc -p tsconfig.test.json && jest --detectOpenHandles",
     "watch": "BABEL_ENV=esm babel --watch src --ignore **/@types/*,**/*.test.tsx,**/*.test.ts --out-dir esm --extensions '.ts,.tsx'",
     "watch-test": "jest --watch"

--- a/packages/react-component-catalog/tsconfig.build.json
+++ b/packages/react-component-catalog/tsconfig.build.json
@@ -1,9 +1,11 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "outDir": "esm",
+    "rootDir": "src",
     "declaration": true,
     "emitDeclarationOnly": true,
-    "incremental": true
+    "incremental": false
   },
   "include": ["src"],
   "exclude": ["**/*.test.ts", "**/*.test.tsx", "node_modules"]

--- a/packages/react-component-catalog/tsconfig.json
+++ b/packages/react-component-catalog/tsconfig.json
@@ -1,8 +1,6 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src",
     /**
      * Composite is required for lernajs monorepos
      * @see https://betterstack.dev/blog/lerna-typescript-monorepo/#heading-extras

